scripts/auto_dock.py

```diff
@@ -143,179 +209,233 @@ def set_docking_state(self, new_docking_state):
-        self.set_action_state('')
+        self.set_action_state("")
         if self.centering_counter >= self.MAX_CENTERING_COUNT:
-            rospy.logwarn('centering failed. reverting to last state: %s', self.last_docking_state)
+            rospy.logwarn(
+                "centering failed. reverting to last state: %s", self.last_docking_state
+            )
             self.aruco_callback_counter = 0
-            self.set_action_state('')
+            self.set_action_state("")
             self.set_docking_state(self.last_docking_state)
             return
         if self.is_in_view:
             [theta, distance, theta_bounds] = self.fid2pos(self.dock_aruco_tf)
-            if abs(theta)>theta_bounds:
+            if abs(theta) > theta_bounds:
                 self.openrover_turn(theta)
             else:
-                rospy.loginfo('centered switching to approach state')
-                self.set_docking_state('approach')
+                rospy.loginfo("centered switching to approach state")
+                self.set_docking_state("approach")
                 self.openrover_stop()
 
     def approach_state_fun(self):
         self.centering_counter = 0
         if self.is_in_view:
             [theta, distance, theta_bounds] = self.fid2pos(self.dock_aruco_tf)
-            if abs(theta)>theta_bounds:
+            if abs(theta) > theta_bounds:
                 rospy.loginfo("approach angle exceeded: %f", abs(theta))
                 self.openrover_stop()
-                self.set_docking_state('centering')
+                self.set_docking_state("centering")
             else:
-                if self.action_state == 'jogging':
+                if self.action_state == "jogging":
+                    # continuing moving forward in 'approach' state using 'jog distance'
+                    # until we get within 'final approach' distance
                     return
                 if abs(distance) < self.FINAL_APPROACH_DISTANCE:
+                    # within final approach distance; move forward remaining distance
                     self.openrover_stop()
-                    self.openrover_forward(2*self.FINAL_APPROACH_DISTANCE)
-                    self.set_docking_state('final_approach')
+                    self.openrover_forward(abs(distance))
+                    self.set_docking_state("final_approach")
                 else:
-                    self.openrover_forward(self.JOG_DISTANCE)
+                    # not yet within final approach distance
+                    # creep forward some distance >= minimum jog distance
+                    # depending on how far away you are
+                    self.openrover_forward(
+                        np.maximum(
+                            (abs(distance) - self.FINAL_APPROACH_DISTANCE) / 2,
+                            self.JOG_DISTANCE,
+                        )
+                    )
         else:
+            # TODO/help - not sure what their logic was here -
```

Comment:
I am not positive, but I think their logic is as follows:
The robot will approach the aruco tag as much as it can but at some point because of the size of aruco tag and position of camera, the aruco tag may be above the camera frame or just outside. At this point, they make the assumption that the robot is as straight as it could possibly be and decides to enter the final approach

Reply:
@michelleat do you think this makes sense?